x-pack/plugins/enterprise_search/server/lib/enterprise_search_config_api.ts

```diff
@@ -135,3 +139,13 @@ export const callEnterpriseSearchConfigAPI = async ({
     clearTimeout(timeout);
   }
 };
+
+export const warnMismatchedVersions = (enterpriseSearchVersion: string, log: Logger) => {
+  const kibanaVersion = kibanaPackageJson.version;
+
+  if (enterpriseSearchVersion !== kibanaVersion) {
+    log.warn(
+      `Your Kibana instance (v${kibanaVersion}) is not the same version as your Enterprise Search instance (v${enterpriseSearchVersion}). Please upgrade your instances accordingly to matching versions, otherwise you may see breaking or otherwise buggy behavior.`
```

Comment:
Oh actually @chriscressman (sorry for bothering you so much today lol), do you want to wordsmith this at all? There's a screenshot of the actual output in the PR description if that helps.

Rereading, I also just realized I used `otherwise` twice in the same clause... ugh.

Reply:
Sure, here you go:

> Your Kibana instance (v8.0.0) is not the same version as your Enterprise Search instance (v7.14.0), which may cause unexpected behavior. Use matching versions for the best experience.

Reply:
Thanks!!
```suggestion
      `Your Kibana instance (v${kibanaVersion}) is not the same version as your Enterprise Search instance (v${enterpriseSearchVersion}), which may cause unexpected behavior. Use matching versions for the best experience.`
```